docs/design/submarine-server/architecture.md

```diff
@@ -0,0 +1,75 @@
+
+## Proposal
+ ```
+                                                               +---------------------+
++-----------+                                                  | +--------+   +----+ |
+|           |                                                  | |runtime1+-->+job1| |
+| workbench +---+    +---------------------------------+       | +--------+   +----+ |
+|           |   |    | +------+ +--------------------+ |   +-->+ +--------+   +----+ |
++-----------+   |    | |      | | +------+ +-------+ | |   |   | |runtime2+-->+job2| |
+                |    | |      | | | YARN | |  K8s  | | |   |   | +--------+   +----+ |
++-----------+   |    | |      | | +------+ +-------+ | |   |   |     YARN Cluster    |
+|           |   |    | |      | |     submitter      | |   |   +---------------------+
+|    CLI    +------->+ |      | +--------------------+ +---+
+|           |   |    | | REST | +--------------------+ |   |   +---------------------+
++-----------+   |    | |      | | +------+ +-------+ | |   |   | +--------+   +----+ |
+                |    | |      | | |AuxMgr| |monitor| | |   |   | |        +-->+job1| |
++-----------+   |    | |      | | +------+ +-------+ | |   |   | |        |   +----+ |
+|           |   |    | |      | |     JobManager     | |   +-->+ |operator|   +----+ |
+|    SDK    +---+    | +------+ +--------------------+ |       | |        +-->+job2| |
+|           |        +---------------------------------+       | +--------+   +----+ |
++-----------+                                                  |     K8s Cluster     |
+   client                          server                      +---------------------+
+ ```
+We propose to split the original core module in the old layout into two modules, CLI and server as shown in FIG. The submarine-client calls the REST APIs to submit and retrieve the job info. The submarine-server provides the REST service, job management, submitting the job to cluster, and running job in different clusters through the corresponding runtime.
+
+## submarine-server
+
+### REST
+The REST API Service handles HTTP requests and is responsible for authentication. It acts as the caller for the JobManager component.
+
+The REST component defines the generic job spec which describes the detailed info about job. For more details, refer to [here](https://docs.google.com/document/d/1kd-5UzsHft6gV7EuZiPXeWIKJtPqVwkNlqMvy0P_pAw/edit#).
+
+### JobManager
+The JobManager receives the job requests, persisting the job metadata in a database(MySQL in production), submitting and monitoring the job. Using the plug-in design pattern for submitter to extends more features. Submitting the job to cluster resource management system through the specified submitter plug-in. 
+
+The JobManager has two main components: auxiliary manager and job monitor.
+
+#### AuxMgr
```

Comment:
To void dependency conflicts, is AuxMgr is mainly used for managing different classloaders of submitter plugins? It's ok for me, but I'm not sure about the name. @liuxunorg, @tangzhankun can you help to take a look at this part?

Reply:
I think `auxiliary ` is a very vague term. maybe the `pluginManger` will be clearer?

Reply:
Agree to adopt the `PluginManager` term which abbr is `PlugMgr`.